core/controller/cluster/controller_operation.go

```diff
@@ -272,7 +274,13 @@ func (c *controller) Rollback(ctx context.Context,
 		return nil, err
 	}
 
-	// 4. rollback cluster config in git repo and update status
+	// 4. update restart time in gitops branch
+	err = c.mergeDefaultBranchIntoGitOps(ctx, application.Name, cluster.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	// 5. rollback cluster config in git repo and update status
 	newConfigCommit, err := c.clusterGitRepo.Rollback(ctx, application.Name, cluster.Name, pipelinerun.ConfigCommit)
```

Comment:
what about a more clear interface?
 Rollback(ctx, application.Name, cluster.Name, branch, to )？
 Rollback(ctx, application.Name, cluster.Name, gitrepo.GitOpsBranch, pielinerun.ConfigCommit )？

Reply:
In principle we only submit commits on gitops branch. How about adding a comment on Rollback?
```go
// Rollback rolls gitOps branch back to a specific commit if there are diffs
Rollback(ctx context.Context, application, cluster, commit string) (string, error)
```